File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> map=new HashMap<>();
        for(String s:strs){
            char p[]=s.toCharArray();
            Arrays.sort(p);
String next = new String(p);
            if(!map.containsKey(next)){
                map.put(next,new ArrayList<String>());
            }
            map.get(next).add(s);
        }
        List<List<String>> ans=new ArrayList<>();
        for(String m:map.keySet()){
            System.out.println(m);
            ans.add(map.get(m));
        }
        return ans;
        
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-0.cs
public class Solution {
    public List<List<string>> GroupAnagrams(string[] strs) {
        Dictionary<string,List<string>> map=new Dictionary<string,List<string>>();
        foreach(string s in strs){
            //haha
            char[] cf=s.ToCharArray();
            Array.Sort(cf);
            string p=new string(cf);
            if(!map.ContainsKey(p)){
                map.Add(p,new List<string>());
            }
            map[p].Add(s);
            
        }
        List<List<string>> ans=new List<List<string>>();
        foreach(var c in map){
            ans.Add(c.Value);
        }
        return ans;

    }
}
